CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/state/DataAccessPreferenceKeys.java

```diff
@@ -69,19 +70,22 @@ public static File getDataAccessResultsDirEx() {
      * <p>
      * Use null to stop writing results.
      * <p>
-     * If the directory is non-null, it will be saved separately so it can be
-     * retrieved and used as a default the next time the user wants to specify a
-     * directory.
+     * If the path is a directory and is non-null, it will be saved separately
+     * so it can be retrieved and used as a default the next time the user wants
+     * to specify a directory. If the file is not a directory, an empty string
+     * will be stored as the directory, and the previous directory will not be
+     * updated.
      *
      * @param dir A directory to write data access results to, or null to not
      * write results.
      */
     public static void setDataAccessResultsDir(final File dir) {
         final Preferences prefs = NbPreferences.forModule(DataAccessPreferenceKeys.class);
-        prefs.put(SAVE_DATA_DIR_PREF, dir == null ? "" : dir.getAbsolutePath());
-        if (dir != null) {
-            prefs.put(PREVIOUS_DATA_DIR_PREF, dir.getAbsolutePath());
+        if (dir != null && dir.isDirectory()) {
+            prefs.put(PREVIOUS_DATA_DIR_PREF, prefs.get(SAVE_DATA_DIR_PREF, ""));
         }
+        prefs.put(SAVE_DATA_DIR_PREF, dir == null || !dir.isDirectory() ? "" : dir.getAbsolutePath());
+
```

Comment:
Added a check to ensure that the dir parameter is actually a dir (consistent with other methods)
Ensured that the previous directory preference was actually going to be saved as the previous (before this change it was storing the same value twice.)